File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> prevMap;

        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];
            if (prevMap.find(diff) != prevMap.end()) {
                return {prevMap[diff], i};
            }
            prevMap.insert({nums[i], i});
        }

        return {};

    }
};

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def bs(start, end) -> int:
            if start > end:
                return -1
            m = start + (end - start) // 2
            if nums[m] == target:
                return m
            if nums[m] < target:
                return bs(m + 1, end)
            else:
                return bs(0, m - 1)
        
        return bs(0, len(nums) - 1)
                